rococo/data/mongodb.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, Dict, List, Tuple, Union
+from rococo.data.base import DbAdapter
+from pymongo import MongoClient
+
+
+class MongoDBAdapter(DbAdapter):
+    """MongoDB adapter for interacting with MongoDB."""
+
+    def __init__(self, mongo_uri: str, mongo_database: str):
+        self.client = MongoClient(mongo_uri)
+        self.db_name = mongo_database
+        self.db = self.client.get_database(mongo_database)
+
+    def __enter__(self):
+        """Context manager entry point for preparing DB connection."""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.db = None
```

Comment:
the best practice is [to close the client when it is not needed](https://www.mongodb.com/community/forums/t/i-am-using-pymongo-do-i-have-to-close-a-mongoclient-after-use/213511/4)